pic-sure-auth-services/src/main/java/edu/harvard/hms/dbmi/avillach/auth/rest/UserService.java

```diff
@@ -442,4 +443,86 @@ private void checkAssociation(List<User> users) throws ProtocolException{
         }
     }
 
+    /**
+     * Create or update a user record, based on the FENCE user profile, which is in JSON format.
+     *
+     * @param node User profile, as it is received from Gen3 FENCE, in JSON format
+     * @return User The actual entity, as it is persisted (if no errors) in the PSAMA database
+     */
+    public User createUserFromFENCEProfile(JsonNode node) {
+        User actual_user = null;
+        try {
+            User user_to_look_up = new User().setSubject("fence|"+node.get("user_id"));
+
+            if (userRepo == null) {
+                throw new RuntimeException("User repo is null");
+
+            }
+
+            actual_user = userRepo.findOrCreate(user_to_look_up);
+
+            actual_user.setSubject("fence|"+node.get("user_id"));
+            actual_user.setEmail(node.get("email").asText());
+            actual_user.setGeneralMetadata(node.asText());
+
+            // Clear current set of roles every time we
+            actual_user.setRoles(new HashSet<>());
+            userRepo.persist(actual_user);
+        } catch (Exception ex) {
```

Comment:
Instead of catching the very top level Exception here, if you don't want to leak any stack trace to the user, my suggestion is catching this at the rest service level.

Reply:
Done. Easier. Throwing it up